src/test/java/uk/gov/mca/beacons/api/services/DeleteBeaconServiceUnitTest.java

```diff
@@ -0,0 +1,121 @@
+package uk.gov.mca.beacons.api.services;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import java.util.UUID;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.mca.beacons.api.domain.AccountHolder;
+import uk.gov.mca.beacons.api.domain.Note;
+import uk.gov.mca.beacons.api.domain.NoteType;
+import uk.gov.mca.beacons.api.dto.DeleteBeaconRequestDTO;
+import uk.gov.mca.beacons.api.exceptions.AccountHolderNotFoundException;
+import uk.gov.mca.beacons.api.gateways.AccountHolderGateway;
+import uk.gov.mca.beacons.api.gateways.BeaconGateway;
+import uk.gov.mca.beacons.api.gateways.NoteGateway;
+
+@ExtendWith(MockitoExtension.class)
+class DeleteBeaconServiceUnitTest {
+
+  @InjectMocks
+  private DeleteBeaconService deleteBeaconService;
+
+  @Mock
+  private BeaconGateway beaconGateway;
+
+  @Mock
+  private AccountHolderGateway accountHolderGateway;
+
+  @Mock
+  private NoteGateway noteGateway;
+
+  @Captor
+  private ArgumentCaptor<Note> noteCaptor;
+
+  @Test
+  void shouldCallTheBeaconGatewayToDeleteTheBeacon() {
+    final var accountHolderId = UUID.randomUUID();
+    final var request = DeleteBeaconRequestDTO
+      .builder()
+      .beaconId(UUID.randomUUID())
+      .accountHolderId(accountHolderId)
+      .reason("Unused on my boat anymore")
+      .build();
+    given(accountHolderGateway.getById(accountHolderId))
+      .willReturn(new AccountHolder());
```

Comment:
It took me a while to work out that the `accountHolderId` refers to the _person requesting the action_.  At first I was thinking "why does the `DeleteBeaconService` need to know the account holder to delete a beacon"?  

Could the `accountHolderId` field be changed to a generic `actorId` or `requesterId` to fit the domain model of "person that is requesting this beacon to be deleted"?

Reply:
I thought about this in general around the account holder vs back office user and in the end I wasn't sure if trying to create a `User` abstraction was going to be more confusing.

We could rename the argument to `actorId` or something similar but as we were calling through to the `AccountHolderGateway::getById` I went with `accountHolderId` but am happy to be swayed on this